### ISSUE TITLE
[InstCombine] reduce (C1 / x) == C2

### ISSUE THREAD

**Issue by rotateright** [llvm/llvm-project#57542]
The tests for https://reviews.llvm.org/D132888 (which is a fix for [#57338](https://github.com/llvm/llvm-project/issues/57338) reveal another missed optimization:
https://alive2.llvm.org/ce/z/-WrjL2

That could be generalized for any constant numerator, any constant compare value, and also a != ("ne") predicate.

We already have a fold for the related pattern with "udiv" and 0 compare constant, so there may be some opportunity to share/extend the code. 

But that fold is probably also too specific and has enhancement potential. For example, it doesn't matter whether this is sdiv or udiv, they both reduce to the same thing:
https://alive2.llvm.org/ce/z/A8uou8

**Comment by Kmeakin** [comment]
This looks like an instance of #55695

**Comment by rotateright** [comment]
Good point - if we handle the pattern with a constant compare operand, then maybe these cases are all handled automatically.

**Comment by sloorush** [comment]
Hey! I would like to work on this issue. Can someone please guide me on where exactly should I start to look into this?

I'm guessing this should be added somewhere in `llvm/lib/Transforms/InstCombine/InstCombineMulDivRem.cpp`.

**Comment by rotateright** [comment]
These patterns end with "icmp" and have an integer division operand, so we're somewhere under InstCombinerImpl::visitICmpInst(). 
foldICmpDivConstant() might work, but it's possible there's some other generalization of the patterns that works better.
